HousingRegisterApi/V1/Services/CSVGeneratorService.cs

```diff
@@ -0,0 +1,115 @@
+using HousingRegisterApi.V1.Domain.FileExport;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace HousingRegisterApi.V1.Services
+{
+    public class CSVGeneratorService : ICSVService
+    {
+        public async Task<byte[]> Generate<T>(T source, CsvParameters csvParameters)
+        {
+            var sourceType = source.GetType();
+            var array = new List<T>();
+
+            if (!sourceType.IsArray)
+            {
+                array.Add(source);
+            }
+            else
+            {
+                array = source as List<T>;
+            }
+
+            return await Task.Run(() =>
+            {
+                return GenerateBytes(array, csvParameters);
+            }).ConfigureAwait(false);
+        }
+
+        private static byte[] GenerateBytes<T>(IList<T> source, CsvParameters csvParameters)
+        {
+            MemoryStream ms = new MemoryStream();
+            using (StreamWriter sw = new StreamWriter(ms))
+            {
+                GetData<T>(source, out List<string> headers, out List<List<object>> rowsOfValues);
+
+                if (csvParameters.IncludeHeaders)
+                {
+                    sw.WriteLine(string.Join(",", headers));
+                }
+
+                foreach (var itemRow in rowsOfValues)
+                {
+                    sw.WriteLine(string.Join(",", itemRow.Select(x => ConvertToString(x))));
+                }
+            };
+
+            return ms.ToArray();
+        }
+
+        private static void GetData<T>(IList<T> array, out List<string> headers, out List<List<object>> rowsOfValues)
+        {
+            headers = GetHeaders(typeof(T));
+            rowsOfValues = new List<List<object>>();
+
+            foreach (var entity in array)
+            {
+                rowsOfValues.Add(GetValues(entity));
+            }
+        }
+
+        private static string ConvertToString(object dataItem)
```

Comment:
In the past, I've used packages like [CsvHelper](https://joshclose.github.io/CsvHelper/) to handle this.